lib/telegraf.py

```diff
@@ -0,0 +1,170 @@
+#
+# [EXPIRIMENTAL]
+#
+# Add Telegraf to an app container to collect StatsD events from the runtime.
+# Metrics will be forwarded to host defined in APPMETRICS_TARGET environment
+# variable which is a JSON string with the following values
+# - url: complate url of the endpoint. Mandatory.
+# - username: basic auth username. Optional.
+# - password: basic auth password. Mandatory if username is specified.
+#
+
+import json
+import os
+
+import buildpackutil
+import datadog
+import subprocess
+from m2ee import logger
+
+
+def _get_appmetrics_target():
+    return os.getenv('APPMETRICS_TARGET')
+
+
+def is_enabled():
+    return _get_appmetrics_target() is not None
+
+
+def _get_appmetrics_target_url():
+    target = json.loads(_get_appmetrics_target())
+    if 'url' not in target:
+        logger.error('APPMETRICS_TARGET.url value is not defined in %s' % _get_appmetrics_target())
+    return target['url']
+
+
+def _get_appmetrics_target_username():
+    target = json.loads(_get_appmetrics_target())
+    return target['username'] if 'username' in target else None
+
+
+def _get_appmetrics_target_password():
+    target = json.loads(_get_appmetrics_target())
+    return target['password'] if 'password' in target else None
+
+
+def _config_value_str(value):
+    if type(value) is str:
+        return '"%s"' % value
+    elif type(value) is int:
+        return value
```

Comment:
Is this correct?

Reply:
yes, for telegraf string values need to be quoted with double quotes.